varcode/coding_effect.py

```diff
@@ -32,12 +35,28 @@
 from .mutate import substitute, insert_after
 from .string_helpers import trim_shared_flanking_strings
 
-from Bio.Seq import Seq
-from Bio.Data import CodonTable
-
+DNA_CODON_TABLE = CodonTable.standard_dna_table.forward_table
 START_CODONS = set(CodonTable.standard_dna_table.start_codons)
 STOP_CODONS = set(CodonTable.standard_dna_table.stop_codons)
 
+def translate_codon(codon, aa_pos):
+    """Translate a single codon into a single amino acid or stop '*'
+
+    Parameters
+    ----------
+    codon : str
+        Expected to be of length 3
+    aa_pos : int
+        Codon/amino acid offset into the protein (starting from 0)
+    """
+    # not handling rare Leucine or Valine starts!
+    if aa_pos == 0 and codon in START_CODONS:
+        return "M"
+    elif codon in STOP_CODONS:
+        return "*"
+    else:
+        return DNA_CODON_TABLE[codon]
+
 def translate(cds_seq):
```

Comment:
Yeah, I have a mild preference for `coding_sequence` but I'd be fine with
`cds`.

On Fri, Mar 27, 2015 at 12:36 PM, Alex Rubinsteyn <notifications@github.com>
wrote:

> In varcode/coding_effect.py
> <https://github.com/hammerlab/varcode/pull/45#discussion_r27309506>:
>
> > +
> > +    Parameters
> > +    ----------
> > +    codon : str
> > +        Expected to be of length 3
> > +    aa_pos : int
> > +        Codon/amino acid offset into the protein (starting from 0)
> > +    """
> > +    # not handling rare Leucine or Valine starts!
> > +    if aa_pos == 0 and codon in START_CODONS:
> > +        return "M"
> > +    elif codon in STOP_CODONS:
> > +        return "*"
> > +    else:
> > +        return DNA_CODON_TABLE[codon]
> > +
> >  def translate(cds_seq):
>
> What do you think of just using cds? It's pretty pervasive genomics and I
> agree that cds_seq is redundant.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/varcode/pull/45/files#r27309506>.
>